Make Page refcounted and replace CheckedPtr with RefPtr for stack variables

cdumez opened the pull request:
#### 23c37146256e9c7382894cda4d929cda2a5b78c6
<pre>
Make Page refcounted and replace CheckedPtr with RefPtr for stack variables
<a href="https://bugs.webkit.org/show_bug.cgi?id=266368">https://bugs.webkit.org/show_bug.cgi?id=266368</a>
<a href="https://rdar.apple.com/118965945">rdar://118965945</a>

Reviewed by Brent Fulgham.

* Source/WebCore/Modules/permissions/MainThreadPermissionObserver.cpp:
(WebCore::MainThreadPermissionObserver::MainThreadPermissionObserver):
* Source/WebCore/Modules/permissions/MainThreadPermissionObserver.h:
* Source/WebCore/Modules/permissions/PermissionController.h:
* Source/WebCore/Modules/permissions/PermissionObserver.h:
* Source/WebCore/Modules/permissions/PermissionStatus.cpp:
(WebCore::PermissionStatus::create):
(WebCore::PermissionStatus::PermissionStatus):
* Source/WebCore/Modules/permissions/PermissionStatus.h:
* Source/WebCore/Scripts/SettingsTemplates/InternalSettingsGenerated.h.erb:
* Source/WebCore/accessibility/atspi/AccessibilityRootAtspi.h:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::MainThreadBridge::ensureOnMainThread):
* Source/WebCore/dom/ConstantPropertyMap.cpp:
(WebCore::ConstantPropertyMap::updateConstantsForSafeAreaInsets):
(WebCore::ConstantPropertyMap::updateConstantsForFullscreen):
* Source/WebCore/dom/DataTransfer.cpp:
(WebCore::DataTransfer::readStringFromPasteboard const):
(WebCore::DataTransfer::setDataFromItemList):
* Source/WebCore/dom/DeviceOrientationAndMotionAccessController.cpp:
(WebCore::DeviceOrientationAndMotionAccessController::shouldAllowAccess):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::configureSharedLogger):
(WebCore::Document::childrenChanged):
(WebCore::Document::setVisualUpdatesAllowed):
(WebCore::Document::willBeRemovedFromFrame):
(WebCore::Document::appHighlightRegistry):
(WebCore::Document::implicitClose):
(WebCore::Document::minimumDOMTimerInterval const):
(WebCore::Document::domTimerAlignmentInterval const):
(WebCore::Document::idbConnectionProxy):
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
(WebCore::Document::dispatchDisabledAdaptationsDidChangeForMainFrame):
(WebCore::Document::viewportArguments const):
(WebCore::Document::updateViewportArguments):
(WebCore::Document::themeColorChanged):
(WebCore::Document::processColorScheme):
(WebCore::Document::processWebAppOrientations):
(WebCore::Document::prepareMouseEvent):
(WebCore::Document::runScrollSteps):
(WebCore::Document::updateIsPlayingMedia):
(WebCore::Document::invalidateRenderingDependentRegions):
(WebCore::Document::setFocusedElement):
(WebCore::Document::cookie):
(WebCore::Document::setCookie):
(WebCore::Document::shouldMaskURLForBindingsInternal const):
(WebCore::Document::setBackForwardCacheState):
(WebCore::Document::suspend):
(WebCore::Document::resume):
(WebCore::Document::registerForCaptionPreferencesChangedCallbacks):
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
(WebCore::Document::postTask):
(WebCore::Document::serviceRequestVideoFrameCallbacks):
(WebCore::Document::exitPointerLock):
(WebCore::Document::requestIdleCallback):
(WebCore::Document::wheelEventHandlersChanged):
(WebCore::Document::deviceScaleFactor const):
(WebCore::Document::useSystemAppearance const):
(WebCore::Document::useDarkAppearance const):
(WebCore::Document::useElevatedUserInterfaceLevel const):
(WebCore::Document::didAssociateFormControl):
(WebCore::Document::didAssociateFormControlsTimerFired):
(WebCore::Document::didLoadResourceSynchronously):
(WebCore::Document::wrapCryptoKey):
(WebCore::Document::unwrapCryptoKey):
(WebCore::Document::hasFocus const):
(WebCore::Document::addPlaybackTargetPickerClient):
(WebCore::Document::removePlaybackTargetPickerClient):
(WebCore::Document::showPlaybackTargetPicker):
(WebCore::Document::playbackTargetPickerClientStateDidChange):
(WebCore::Document::scheduleRenderingUpdate):
(WebCore::Document::logger):
(WebCore::Document::didLogMessage):
(WebCore::Document::notificationClient):
(WebCore::Document::graphicsClient):
(WebCore::Document::sessionID const):
(WebCore::Document::noiseInjectionHashSalt const):
(WebCore::Document::mediaKeysStorageDirectory):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::DocumentMarkerController::invalidateRectsForAllMarkers):
(WebCore::DocumentMarkerController::invalidateRectsForMarkersInNode):
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::hasStorageAccess):
(WebCore::DocumentStorageAccess::requestStorageAccess):
(WebCore::DocumentStorageAccess::requestStorageAccessQuirk):
* Source/WebCore/dom/Element.cpp:
(WebCore::dispatchPointerEventIfNeeded):
(WebCore::Element::removedFromAncestor):
(WebCore::Element::focus):
(WebCore::Element::dispatchFocusEvent):
(WebCore::Element::dispatchBlurEvent):
(WebCore::Element::setPointerCapture):
(WebCore::Element::releasePointerCapture):
(WebCore::Element::hasPointerCapture):
(WebCore::Element::requestPointerLock):
* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::ExtensionStyleSheets::updateInjectedStyleSheetCache const):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::removeOverlaySoonIfNeeded):
(WebCore::ImageOverlay::updateWithTextRecognitionResult):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::inspect):
* Source/WebCore/dom/PseudoElement.cpp:
(WebCore::PseudoElement::create):
(WebCore::PseudoElement::clearHostElement):
* Source/WebCore/dom/ScriptedAnimationController.cpp:
(WebCore::ScriptedAnimationController::registerCallback):
(WebCore::ScriptedAnimationController::scheduleAnimation):
* Source/WebCore/dom/VisitedLinkState.cpp:
(WebCore::VisitedLinkState::determineLinkStateSlowCase):
* Source/WebCore/dom/WindowEventLoop.h:
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
(WebCore::ImageControlsMac::handleEvent):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::ReplacementFragment):
* Source/WebCore/editing/SelectionGeometryGatherer.cpp:
(WebCore::SelectionGeometryGatherer::Notifier::~Notifier):
* Source/WebCore/editing/SpellChecker.cpp:
(WebCore::SpellChecker::client const):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::sanitizeMarkupWithArchive):
(WebCore::WebContentReader::readPlainText):
* Source/WebCore/editing/markup.cpp:
(WebCore::createPageForSanitizingWebContent):
(WebCore::sanitizeMarkup):
* Source/WebCore/editing/markup.h:
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::attributeChanged):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::setRecalcListItems):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/inspector/PageDebugger.cpp:
(WebCore::PageDebugger::runEventLoopWhilePausedInternal):
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::DocumentThreadableLoader):
(WebCore::DocumentThreadableLoader::makeCrossOriginAccessRequest):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::ForbidPromptsScope::ForbidPromptsScope):
(WebCore::ForbidPromptsScope::~ForbidPromptsScope):
(WebCore::ForbidSynchronousLoadsScope::ForbidSynchronousLoadsScope):
(WebCore::ForbidSynchronousLoadsScope::~ForbidSynchronousLoadsScope):
(WebCore::FrameLoader::FrameProgressTracker::progressCompleted):
(WebCore::FrameLoader::checkCompletenessNow):
(WebCore::FrameLoader::setOpener):
(WebCore::FrameLoader::provisionalLoadStarted):
(WebCore::FrameLoader::loadFrameRequest):
(WebCore::FrameLoader::loadWithDocumentLoader):
(WebCore::FrameLoader::setState):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::shouldClose):
(WebCore::FrameLoader::dispatchBeforeUnloadEvent):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::continueLoadAfterNewWindowPolicy):
(WebCore::FrameLoader::loadedResourceFromMemoryCache):
(WebCore::FrameLoader::dispatchDidClearWindowObjectInWorld):
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
(WebCore::createWindow):
(WebCore::FrameLoader::switchBrowsingContextsGroup):
* Source/WebCore/loader/SubframeLoader.cpp:
(WebCore::FrameLoader::SubframeLoader::requestObject):
(WebCore::FrameLoader::SubframeLoader::loadOrRedirectSubframe):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::canMouseDownStartSelect):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::updateDragSourceActionsAllowed const):
(WebCore::EventHandler::startAutoHideCursorTimer):
(WebCore::EventHandler::autoHideCursorTimerFired):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::mouseMoved):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::updateMouseEventTargetNode):
(WebCore::EventHandler::clearElementUnderMouse):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::clearLatchedState):
(WebCore::EventHandler::scheduleCursorUpdate):
(WebCore::EventHandler::dispatchFakeMouseMoveEventSoon):
(WebCore::EventHandler::textRecognitionHoverTimerFired):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::dragCancelled):
(WebCore::EventHandler::dragSourceEndedAt):
(WebCore::EventHandler::dispatchDragStartEventOnSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::tabsToLinks const):
(WebCore::EventHandler::defaultBackspaceEventHandler):
(WebCore::EventHandler::beginKeyboardScrollGesture):
(WebCore::EventHandler::defaultArrowEventHandler):
(WebCore::EventHandler::defaultTabEventHandler):
(WebCore::EventHandler::focusDocumentView):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::canShowModalDialog):
(WebCore::LocalDOMWindow::prewarmLocalStorageIfNecessary):
(WebCore::LocalDOMWindow::protectedPage const):
(WebCore::LocalDOMWindow::shouldHaveWebKitNamespaceForWorld):
(WebCore::LocalDOMWindow::webkitNamespace):
(WebCore::LocalDOMWindow::focus):
(WebCore::LocalDOMWindow::blur):
(WebCore::LocalDOMWindow::close):
(WebCore::LocalDOMWindow::print):
(WebCore::LocalDOMWindow::alert):
(WebCore::LocalDOMWindow::confirmForBindings):
(WebCore::LocalDOMWindow::prompt):
(WebCore::LocalDOMWindow::outerHeight const):
(WebCore::LocalDOMWindow::outerWidth const):
(WebCore::LocalDOMWindow::screenX const):
(WebCore::LocalDOMWindow::screenY const):
(WebCore::LocalDOMWindow::closed const):
(WebCore::LocalDOMWindow::devicePixelRatio const):
(WebCore::LocalDOMWindow::moveBy const):
(WebCore::LocalDOMWindow::moveTo const):
(WebCore::LocalDOMWindow::resizeBy const):
(WebCore::LocalDOMWindow::resizeTo const):
(WebCore::LocalDOMWindow::deviceOrientationController const):
(WebCore::LocalDOMWindow::deviceMotionController const):
(WebCore::LocalDOMWindow::incrementScrollEventListenersCount):
(WebCore::LocalDOMWindow::decrementScrollEventListenersCount):
(WebCore::LocalDOMWindow::createWindow):
(WebCore::LocalDOMWindow::open):
(WebCore::LocalDOMWindow::enableSuddenTermination):
(WebCore::LocalDOMWindow::disableSuddenTermination):
(WebCore::LocalDOMWindow::checkedPage const): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::setDocument):
(WebCore::LocalFrame::orientation const):
(WebCore::LocalFrame::injectUserScripts):
(WebCore::LocalFrame::willDetachPage):
(WebCore::LocalFrame::setPageAndTextZoomFactors):
(WebCore::LocalFrame::frameScaleFactor const):
(WebCore::LocalFrame::screenSize const):
(WebCore::LocalFrame::documentURLDidChange):
* Source/WebCore/page/OpportunisticTaskScheduler.cpp:
(WebCore::OpportunisticTaskScheduler::rescheduleIfNeeded):
(WebCore::OpportunisticTaskScheduler::runLoopObserverFired):
(WebCore::OpportunisticTaskScheduler::checkedPage const): Deleted.
* Source/WebCore/page/OpportunisticTaskScheduler.h:
* Source/WebCore/page/Page.cpp:
(WebCore::allPages):
(WebCore::Page::forEachPage):
(WebCore::networkStateChanged):
(WebCore::createMainFrame):
(WebCore::Page::create):
(WebCore::m_historyItemClient):
(WebCore::Page::~Page):
(WebCore::Page::incrementNestedRunLoopCount): Deleted.
(WebCore::Page::decrementNestedRunLoopCount): Deleted.
(WebCore::Page::whenUnnested): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Frame::protectedPage const):
(WebCore::Document::protectedPage const):
(WebCore::Page::insideNestedRunLoop const): Deleted.
(WebCore::Frame::checkedPage const): Deleted.
(WebCore::Document::checkedPage const): Deleted.
* Source/WebCore/page/PageGroup.h:
(WebCore::PageGroup::pages const):
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/UserContentProvider.h:
* Source/WebCore/page/VisitedLinkStore.h:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::wheelEvent):
(WebCore::EventHandler::tabsToAllFormControls const):
(WebCore::EventHandler::focusDocumentView):
(WebCore::EventHandler::passMouseDownEventToWidget):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::wheelEvent):
(WebCore::EventHandler::focusDocumentView):
(WebCore::EventHandler::passMouseDownEventToWidget):
(WebCore::EventHandler::currentPlatformMouseEvent const):
(WebCore::EventHandler::tabsToAllFormControls const):
(WebCore::EventHandler::determineWheelEventTarget):
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/platform/mac/DataDetectorHighlight.h:
* Source/WebCore/plugins/PluginInfoProvider.h:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::scheduleRenderingUpdateForImage):
* Source/WebCore/svg/graphics/SVGImage.cpp:
(WebCore::SVGImage::dataChanged):
* Source/WebCore/svg/graphics/SVGImage.h:
* Source/WebCore/testing/InternalSettings.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::ServiceWorkerThreadProxy):
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.h:
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.cpp:
(WebCore::SharedWorkerThreadProxy::SharedWorkerThreadProxy):
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::jsPDFDocPrint):
(WebKit::PDFPlugin::paintControlForLayerInContext):
(WebKit::PDFPlugin::convertFromPDFViewToScreen const):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::boundsOnScreen const):
(WebKit::PDFPluginBase::isActive const):
(WebKit::PDFPluginBase::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebKit::PDFPluginBase::deviceScaleFactor const):
(WebKit::PDFPluginBase::createScrollbar):
(WebKit::PDFPluginBase::hudEnabled const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::teardown):
(WebKit::UnifiedPDFPlugin::createGraphicsLayer):
(WebKit::UnifiedPDFPlugin::scheduleRenderingUpdate):
(WebKit::UnifiedPDFPlugin::updateLayerHierarchy):
(WebKit::UnifiedPDFPlugin::didChangeSettings):
(WebKit::UnifiedPDFPlugin::didChangeIsInWindow):
(WebKit::UnifiedPDFPlugin::pageScaleFactor const):
(WebKit::UnifiedPDFPlugin::setPageScaleFactor):
(WebKit::UnifiedPDFPlugin::updateScrollingExtents):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp:
(WebKit::WebPermissionController::query):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
(WebKit::WebPage::close):
(WebKit::DeferredPageDestructor::createDeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::DeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::tryDestruction): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.h:
* Source/WebKitLegacy/WebCoreSupport/PageStorageSessionProvider.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.mm:
(WebMediaPlaybackTargetPicker::WebMediaPlaybackTargetPicker):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView _didConcludeEditDrag]):
(-[WebView _close]):
(-[WebView page]):
(-[WebView _checkDidPerformFirstNavigation]):
(-[WebView _executeCoreCommandByName:value:]):
(-[WebView _geolocationDidChangePosition:]):
(-[WebView _geolocationDidFailWithMessage:]):
(-[WebView _notificationIDForTesting:]):
(-[WebView _clearNotificationPermissionState]):
(WebKit::DeferredPageDestructor::createDeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::DeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::tryDestruction): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:

Canonical link: <a href="https://commits.webkit.org/272060@main">https://commits.webkit.org/272060@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/17b6150d08a388e6ec68fa8f11b499046c97a4b3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30365 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9039 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31974 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32872 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27472 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31065 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11264 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27431 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30673 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7587 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27247 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6501 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6647 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27066 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34211 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27678 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27567 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32838 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6628 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4781 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30656 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8386 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7378 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3951 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7168 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -2875,7 +2875,8 @@ void Document::willBeRemovedFromFrame()
 
-    if (CheckedPtr page = this->page()) {
+    // Unable to ref the page as it may have started destruction.
+    if (WeakPtr page = this->page()) {
```
Should this be `SingleThreadWeakPtr`?

cdumez replied:
No need, with C++17, we don't need to specify template parameters. SingleThreadWeakPtr is only an alias to WeakPtr with specific template parameters.

brentfulgham reviewed (approved):
Looks great!

cdumez committed:
Make Page refcounted and replace CheckedPtr with RefPtr for stack variables
https://bugs.webkit.org/show_bug.cgi?id=266368
rdar://118965945

Reviewed by Brent Fulgham.

* Source/WebCore/Modules/permissions/MainThreadPermissionObserver.cpp:
(WebCore::MainThreadPermissionObserver::MainThreadPermissionObserver):
* Source/WebCore/Modules/permissions/MainThreadPermissionObserver.h:
* Source/WebCore/Modules/permissions/PermissionController.h:
* Source/WebCore/Modules/permissions/PermissionObserver.h:
* Source/WebCore/Modules/permissions/PermissionStatus.cpp:
(WebCore::PermissionStatus::create):
(WebCore::PermissionStatus::PermissionStatus):
* Source/WebCore/Modules/permissions/PermissionStatus.h:
* Source/WebCore/Scripts/SettingsTemplates/InternalSettingsGenerated.h.erb:
* Source/WebCore/accessibility/atspi/AccessibilityRootAtspi.h:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::initScriptForWindowProxy):
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::MainThreadBridge::ensureOnMainThread):
* Source/WebCore/dom/ConstantPropertyMap.cpp:
(WebCore::ConstantPropertyMap::updateConstantsForSafeAreaInsets):
(WebCore::ConstantPropertyMap::updateConstantsForFullscreen):
* Source/WebCore/dom/DataTransfer.cpp:
(WebCore::DataTransfer::readStringFromPasteboard const):
(WebCore::DataTransfer::setDataFromItemList):
* Source/WebCore/dom/DeviceOrientationAndMotionAccessController.cpp:
(WebCore::DeviceOrientationAndMotionAccessController::shouldAllowAccess):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::configureSharedLogger):
(WebCore::Document::childrenChanged):
(WebCore::Document::setVisualUpdatesAllowed):
(WebCore::Document::willBeRemovedFromFrame):
(WebCore::Document::appHighlightRegistry):
(WebCore::Document::implicitClose):
(WebCore::Document::minimumDOMTimerInterval const):
(WebCore::Document::domTimerAlignmentInterval const):
(WebCore::Document::idbConnectionProxy):
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
(WebCore::Document::dispatchDisabledAdaptationsDidChangeForMainFrame):
(WebCore::Document::viewportArguments const):
(WebCore::Document::updateViewportArguments):
(WebCore::Document::themeColorChanged):
(WebCore::Document::processColorScheme):
(WebCore::Document::processWebAppOrientations):
(WebCore::Document::prepareMouseEvent):
(WebCore::Document::runScrollSteps):
(WebCore::Document::updateIsPlayingMedia):
(WebCore::Document::invalidateRenderingDependentRegions):
(WebCore::Document::setFocusedElement):
(WebCore::Document::cookie):
(WebCore::Document::setCookie):
(WebCore::Document::shouldMaskURLForBindingsInternal const):
(WebCore::Document::setBackForwardCacheState):
(WebCore::Document::suspend):
(WebCore::Document::resume):
(WebCore::Document::registerForCaptionPreferencesChangedCallbacks):
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
(WebCore::Document::postTask):
(WebCore::Document::serviceRequestVideoFrameCallbacks):
(WebCore::Document::exitPointerLock):
(WebCore::Document::requestIdleCallback):
(WebCore::Document::wheelEventHandlersChanged):
(WebCore::Document::deviceScaleFactor const):
(WebCore::Document::useSystemAppearance const):
(WebCore::Document::useDarkAppearance const):
(WebCore::Document::useElevatedUserInterfaceLevel const):
(WebCore::Document::didAssociateFormControl):
(WebCore::Document::didAssociateFormControlsTimerFired):
(WebCore::Document::didLoadResourceSynchronously):
(WebCore::Document::wrapCryptoKey):
(WebCore::Document::unwrapCryptoKey):
(WebCore::Document::hasFocus const):
(WebCore::Document::addPlaybackTargetPickerClient):
(WebCore::Document::removePlaybackTargetPickerClient):
(WebCore::Document::showPlaybackTargetPicker):
(WebCore::Document::playbackTargetPickerClientStateDidChange):
(WebCore::Document::scheduleRenderingUpdate):
(WebCore::Document::logger):
(WebCore::Document::didLogMessage):
(WebCore::Document::notificationClient):
(WebCore::Document::graphicsClient):
(WebCore::Document::sessionID const):
(WebCore::Document::noiseInjectionHashSalt const):
(WebCore::Document::mediaKeysStorageDirectory):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::DocumentMarkerController::invalidateRectsForAllMarkers):
(WebCore::DocumentMarkerController::invalidateRectsForMarkersInNode):
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::hasStorageAccess):
(WebCore::DocumentStorageAccess::requestStorageAccess):
(WebCore::DocumentStorageAccess::requestStorageAccessQuirk):
* Source/WebCore/dom/Element.cpp:
(WebCore::dispatchPointerEventIfNeeded):
(WebCore::Element::removedFromAncestor):
(WebCore::Element::focus):
(WebCore::Element::dispatchFocusEvent):
(WebCore::Element::dispatchBlurEvent):
(WebCore::Element::setPointerCapture):
(WebCore::Element::releasePointerCapture):
(WebCore::Element::hasPointerCapture):
(WebCore::Element::requestPointerLock):
* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::ExtensionStyleSheets::updateInjectedStyleSheetCache const):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::removeOverlaySoonIfNeeded):
(WebCore::ImageOverlay::updateWithTextRecognitionResult):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::inspect):
* Source/WebCore/dom/PseudoElement.cpp:
(WebCore::PseudoElement::create):
(WebCore::PseudoElement::clearHostElement):
* Source/WebCore/dom/ScriptedAnimationController.cpp:
(WebCore::ScriptedAnimationController::registerCallback):
(WebCore::ScriptedAnimationController::scheduleAnimation):
* Source/WebCore/dom/VisitedLinkState.cpp:
(WebCore::VisitedLinkState::determineLinkStateSlowCase):
* Source/WebCore/dom/WindowEventLoop.h:
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
(WebCore::ImageControlsMac::handleEvent):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::ReplacementFragment):
* Source/WebCore/editing/SelectionGeometryGatherer.cpp:
(WebCore::SelectionGeometryGatherer::Notifier::~Notifier):
* Source/WebCore/editing/SpellChecker.cpp:
(WebCore::SpellChecker::client const):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::sanitizeMarkupWithArchive):
(WebCore::WebContentReader::readPlainText):
* Source/WebCore/editing/markup.cpp:
(WebCore::createPageForSanitizingWebContent):
(WebCore::sanitizeMarkup):
* Source/WebCore/editing/markup.h:
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::attributeChanged):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::setRecalcListItems):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/inspector/PageDebugger.cpp:
(WebCore::PageDebugger::runEventLoopWhilePausedInternal):
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::DocumentThreadableLoader):
(WebCore::DocumentThreadableLoader::makeCrossOriginAccessRequest):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::ForbidPromptsScope::ForbidPromptsScope):
(WebCore::ForbidPromptsScope::~ForbidPromptsScope):
(WebCore::ForbidSynchronousLoadsScope::ForbidSynchronousLoadsScope):
(WebCore::ForbidSynchronousLoadsScope::~ForbidSynchronousLoadsScope):
(WebCore::FrameLoader::FrameProgressTracker::progressCompleted):
(WebCore::FrameLoader::checkCompletenessNow):
(WebCore::FrameLoader::setOpener):
(WebCore::FrameLoader::provisionalLoadStarted):
(WebCore::FrameLoader::loadFrameRequest):
(WebCore::FrameLoader::loadWithDocumentLoader):
(WebCore::FrameLoader::setState):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::shouldClose):
(WebCore::FrameLoader::dispatchBeforeUnloadEvent):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::continueLoadAfterNewWindowPolicy):
(WebCore::FrameLoader::loadedResourceFromMemoryCache):
(WebCore::FrameLoader::dispatchDidClearWindowObjectInWorld):
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
(WebCore::createWindow):
(WebCore::FrameLoader::switchBrowsingContextsGroup):
* Source/WebCore/loader/SubframeLoader.cpp:
(WebCore::FrameLoader::SubframeLoader::requestObject):
(WebCore::FrameLoader::SubframeLoader::loadOrRedirectSubframe):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::canMouseDownStartSelect):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::updateDragSourceActionsAllowed const):
(WebCore::EventHandler::startAutoHideCursorTimer):
(WebCore::EventHandler::autoHideCursorTimerFired):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::mouseMoved):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::updateMouseEventTargetNode):
(WebCore::EventHandler::clearElementUnderMouse):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::clearLatchedState):
(WebCore::EventHandler::scheduleCursorUpdate):
(WebCore::EventHandler::dispatchFakeMouseMoveEventSoon):
(WebCore::EventHandler::textRecognitionHoverTimerFired):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::dragCancelled):
(WebCore::EventHandler::dragSourceEndedAt):
(WebCore::EventHandler::dispatchDragStartEventOnSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::tabsToLinks const):
(WebCore::EventHandler::defaultBackspaceEventHandler):
(WebCore::EventHandler::beginKeyboardScrollGesture):
(WebCore::EventHandler::defaultArrowEventHandler):
(WebCore::EventHandler::defaultTabEventHandler):
(WebCore::EventHandler::focusDocumentView):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::canShowModalDialog):
(WebCore::LocalDOMWindow::prewarmLocalStorageIfNecessary):
(WebCore::LocalDOMWindow::protectedPage const):
(WebCore::LocalDOMWindow::shouldHaveWebKitNamespaceForWorld):
(WebCore::LocalDOMWindow::webkitNamespace):
(WebCore::LocalDOMWindow::focus):
(WebCore::LocalDOMWindow::blur):
(WebCore::LocalDOMWindow::close):
(WebCore::LocalDOMWindow::print):
(WebCore::LocalDOMWindow::alert):
(WebCore::LocalDOMWindow::confirmForBindings):
(WebCore::LocalDOMWindow::prompt):
(WebCore::LocalDOMWindow::outerHeight const):
(WebCore::LocalDOMWindow::outerWidth const):
(WebCore::LocalDOMWindow::screenX const):
(WebCore::LocalDOMWindow::screenY const):
(WebCore::LocalDOMWindow::closed const):
(WebCore::LocalDOMWindow::devicePixelRatio const):
(WebCore::LocalDOMWindow::moveBy const):
(WebCore::LocalDOMWindow::moveTo const):
(WebCore::LocalDOMWindow::resizeBy const):
(WebCore::LocalDOMWindow::resizeTo const):
(WebCore::LocalDOMWindow::deviceOrientationController const):
(WebCore::LocalDOMWindow::deviceMotionController const):
(WebCore::LocalDOMWindow::incrementScrollEventListenersCount):
(WebCore::LocalDOMWindow::decrementScrollEventListenersCount):
(WebCore::LocalDOMWindow::createWindow):
(WebCore::LocalDOMWindow::open):
(WebCore::LocalDOMWindow::enableSuddenTermination):
(WebCore::LocalDOMWindow::disableSuddenTermination):
(WebCore::LocalDOMWindow::checkedPage const): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::setDocument):
(WebCore::LocalFrame::orientation const):
(WebCore::LocalFrame::injectUserScripts):
(WebCore::LocalFrame::willDetachPage):
(WebCore::LocalFrame::setPageAndTextZoomFactors):
(WebCore::LocalFrame::frameScaleFactor const):
(WebCore::LocalFrame::screenSize const):
(WebCore::LocalFrame::documentURLDidChange):
* Source/WebCore/page/OpportunisticTaskScheduler.cpp:
(WebCore::OpportunisticTaskScheduler::rescheduleIfNeeded):
(WebCore::OpportunisticTaskScheduler::runLoopObserverFired):
(WebCore::OpportunisticTaskScheduler::checkedPage const): Deleted.
* Source/WebCore/page/OpportunisticTaskScheduler.h:
* Source/WebCore/page/Page.cpp:
(WebCore::allPages):
(WebCore::Page::forEachPage):
(WebCore::networkStateChanged):
(WebCore::createMainFrame):
(WebCore::Page::create):
(WebCore::m_historyItemClient):
(WebCore::Page::~Page):
(WebCore::Page::incrementNestedRunLoopCount): Deleted.
(WebCore::Page::decrementNestedRunLoopCount): Deleted.
(WebCore::Page::whenUnnested): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Frame::protectedPage const):
(WebCore::Document::protectedPage const):
(WebCore::Page::insideNestedRunLoop const): Deleted.
(WebCore::Frame::checkedPage const): Deleted.
(WebCore::Document::checkedPage const): Deleted.
* Source/WebCore/page/PageGroup.h:
(WebCore::PageGroup::pages const):
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/UserContentProvider.h:
* Source/WebCore/page/VisitedLinkStore.h:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::wheelEvent):
(WebCore::EventHandler::tabsToAllFormControls const):
(WebCore::EventHandler::focusDocumentView):
(WebCore::EventHandler::passMouseDownEventToWidget):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::wheelEvent):
(WebCore::EventHandler::focusDocumentView):
(WebCore::EventHandler::passMouseDownEventToWidget):
(WebCore::EventHandler::currentPlatformMouseEvent const):
(WebCore::EventHandler::tabsToAllFormControls const):
(WebCore::EventHandler::determineWheelEventTarget):
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/platform/mac/DataDetectorHighlight.h:
* Source/WebCore/plugins/PluginInfoProvider.h:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::scheduleRenderingUpdateForImage):
* Source/WebCore/svg/graphics/SVGImage.cpp:
(WebCore::SVGImage::dataChanged):
* Source/WebCore/svg/graphics/SVGImage.h:
* Source/WebCore/testing/InternalSettings.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::ServiceWorkerThreadProxy):
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.h:
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.cpp:
(WebCore::SharedWorkerThreadProxy::SharedWorkerThreadProxy):
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::jsPDFDocPrint):
(WebKit::PDFPlugin::paintControlForLayerInContext):
(WebKit::PDFPlugin::convertFromPDFViewToScreen const):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::boundsOnScreen const):
(WebKit::PDFPluginBase::isActive const):
(WebKit::PDFPluginBase::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebKit::PDFPluginBase::deviceScaleFactor const):
(WebKit::PDFPluginBase::createScrollbar):
(WebKit::PDFPluginBase::hudEnabled const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::teardown):
(WebKit::UnifiedPDFPlugin::createGraphicsLayer):
(WebKit::UnifiedPDFPlugin::scheduleRenderingUpdate):
(WebKit::UnifiedPDFPlugin::updateLayerHierarchy):
(WebKit::UnifiedPDFPlugin::didChangeSettings):
(WebKit::UnifiedPDFPlugin::didChangeIsInWindow):
(WebKit::UnifiedPDFPlugin::pageScaleFactor const):
(WebKit::UnifiedPDFPlugin::setPageScaleFactor):
(WebKit::UnifiedPDFPlugin::updateScrollingExtents):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp:
(WebKit::WebPermissionController::query):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
(WebKit::WebPage::close):
(WebKit::DeferredPageDestructor::createDeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::DeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::tryDestruction): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKitLegacy/Storage/WebStorageNamespaceProvider.h:
* Source/WebKitLegacy/WebCoreSupport/PageStorageSessionProvider.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.mm:
(WebMediaPlaybackTargetPicker::WebMediaPlaybackTargetPicker):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView _didConcludeEditDrag]):
(-[WebView _close]):
(-[WebView page]):
(-[WebView _checkDidPerformFirstNavigation]):
(-[WebView _executeCoreCommandByName:value:]):
(-[WebView _geolocationDidChangePosition:]):
(-[WebView _geolocationDidFailWithMessage:]):
(-[WebView _notificationIDForTesting:]):
(-[WebView _clearNotificationPermissionState]):
(WebKit::DeferredPageDestructor::createDeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::DeferredPageDestructor): Deleted.
(WebKit::DeferredPageDestructor::tryDestruction): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:

Canonical link: https://commits.webkit.org/272060@main
